messaging/src/main/java/org/axonframework/eventhandling/processors/streaming/pooled/MergeTask.java

```diff
@@ -156,9 +159,14 @@ private Boolean mergeSegments(Segment thisSegment, TrackingToken thisToken,
                 unitOfWorkFactory
                         .create()
                         .executeWithResult(context -> {
-                            tokenStore.deleteToken(name, tokenToDelete);
-                            tokenStore.storeToken(mergedToken, name, mergedSegment.getSegmentId());
-                            tokenStore.releaseClaim(name, mergedSegment.getSegmentId());
+                            joinAndUnwrap(tokenStore.deleteToken(name, tokenToDelete, null));
+                            joinAndUnwrap(tokenStore.storeToken(mergedToken,
+                                                                name,
+                                                                mergedSegment.getSegmentId(),
+                                                                context));
+                            joinAndUnwrap(tokenStore.releaseClaim(name,
+                                                                  mergedSegment.getSegmentId(),
+                                                                  null));
```

Comment:
```suggestion
                            joinAndUnwrap(tokenStore.deleteToken(name, tokenToDelete, context));
                            joinAndUnwrap(tokenStore.storeToken(mergedToken,
                                                                name,
                                                                mergedSegment.getSegmentId(),
                                                                context));
                            joinAndUnwrap(tokenStore.releaseClaim(name,
                                                                  mergedSegment.getSegmentId(),
                                                                  context));
```

We should pass the `ProcessingContext` here.